ng2-components/ng2-alfresco-core/src/directives/node-delete.directive.ts

```diff
@@ -0,0 +1,227 @@
+/*!
+ * @license
+ * Copyright 2016 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Directive, EventEmitter, HostListener, Input, Output } from '@angular/core';
+import { MinimalNodeEntity } from 'alfresco-js-api';
+import { Observable } from 'rxjs/Rx';
+import { AlfrescoApiService } from '../services/alfresco-api.service';
+import { NotificationService } from '../services/notification.service';
+import { TranslationService } from '../services/translation.service';
+
+@Directive({
+    selector: '[adf-delete]',
+    exportAs: 'adfDelete'
+})
+export class NodeDeleteDirective {
+    private nodesApi;
+
+    @Input('adf-delete')
+    selection: MinimalNodeEntity[];
+
+    @Input() permanent: boolean = false;
+
+    @Output() delete: EventEmitter<any> = new EventEmitter();
+
+    @HostListener('click')
+    onClick() {
+        this.process(this.selection);
+    }
+
+    constructor(
+        private notification: NotificationService,
+        private alfrescoApiService: AlfrescoApiService,
+        private translation: TranslationService
+    ) {
+        this.nodesApi = this.alfrescoApiService.getInstance().nodes;
+    }
+
+    process(selection: any, isRestore = false) {
```

Comment:
- avoid to use boolean in method firm . Two behaviour in the same class are bad pratice 
https://martinfowler.com/bliki/FlagArgument.html.
- Is possible to use a type instead of any?